app/services/courses/generate_course_title_service.rb

```diff
@@ -0,0 +1,54 @@
+module Courses
+  class GenerateCourseTitleService
+    def execute(course:)
+      subjects = course.subjects
+
+      title = if course.further_education_course?
+                further_education_title
+              elsif is_modern_lanuage_course?(subjects)
+                modern_language_title(subjects)
+              else
+                generated_title(subjects)
+              end
+
+      title = append_send_info(title) if course.is_send?
+      title
+    end
+
+  private
+
+    def is_modern_lanuage_course?(subjects)
+      subjects[0] == SecondarySubject.modern_languages
+    end
+
+    def modern_language_title(subjects)
```

Comment:
outcome will look amazing wrong....

https://github.com/DFE-Digital/manage-courses-backend/blob/4a627e7c3e15dac86a87f605cfdf0fb5ef6fd94a/app/services/subject_creator_service.rb#L56

Reply:
@defong , trying to decypher what you mean exactly, are you talking about the presence of "English as a second or other language" here? Or about the fact that, again, `subjects` could in theory include more than just "Modern Languages" + languages in it?

Reply:
@misaka 
the output
ie
```ruby
Modern Languages (#{subject_name})
```

throw out

`Modern Languages (Modern languages (other))`
`Modern Languages (English as a second or other language)`

in whichever single, duel, or triple form

Reply:
I'll chat with @fofr about this tomorrow :eyes:

Reply:
> Modern Languages (English as a second or other language)
Could probably shorten to "Modern Languages (English)"

Reply:
Yeah that makes sense - not the nicest having a bunch of edge case handling but unless we do a database migration to update the name (which would have other visual downsides anyway) it'll have to do for now :thinking: